### PR TITLE
Fix coding style, devDependencies, dependencies and ignore common files

### PR DESCRIPTION
**Pull request by dcdeiv**
* FIX
  * Coding style: fix style according to the [jQuery style guide](https://contribute.jquery.org/style-guide/js/).
  * Remove unnecessary "grunt-cli" from `devDependencies` in the `package.json` file. `grunt-cli` must be installed globally via `[sudo] npm install -g grunt-cli` as the [getting-started](http://gruntjs.com/getting-started) guide shows.
  * Add *jQuery* as dependency of npm and bower package.
  * Ignore **npm-debug.log**, **bower_components** and **node_modules**.
    * *bower_components/* will store the jQuery package. It must be ignored in `.gitignore`, `bower.json`, and `.npmignore` because:
      1. **.gitignore**: It must not be uploaded on the git repository. Dependecies must be installed locally via Bower;
      2. **.npmignore**: It is already a node dependency, so it will be installed with npm;
      1. **bower.json**: It will be installed automatically by bower in the `bower_components/` folder later and this folder won't be necessarily in the plugin folder.
    * *node_module* must be ignored by git and Bower because of grunt. Node modules must be stored locally and then installed via npm.

### CODE REVIEW

File: LICENSE-MIT [hunk 1]
```diff
@@ -0,0 +1,23 @@
+
+Copyright (c) 2015 Davide Di Criscito.
```
Comment (heldr) [inline LICENSE-MIT]: Uhm...in this case 2016 Zeno Rocha and we don't need the file since MIT license is on Readme.

File: bower.json [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
-	"license": "MIT"
+    "jquery",
+    "plugin",
+    "boilerplate",
+    "jquery-plugin",
+    "jquery-boilerplate"
+  ],
+  "main": "src/jquery.boilerplate.js",
+	"dependencies": {
+    "jquery": "1"
```
Comment (heldr) [inline bower.json]: as per package.json version >= 1.5.1

File: package.json [hunk 3]
```diff
@@ -40,11 +40,10 @@
   ],
   "license": "MIT",
   "dependencies": {
-    "jquery": ">= 1.5.1"
+    "jquery": "^1.12.0"
```
Comment (heldr) [inline package.json]: >= 1.5.1

File: bower.json [hunk 4]
```diff
@@ -1,18 +1,26 @@
...
+  },
+  "ignore": [
+    "npm-debug.log",
+		"bower_components",
```
Comment (heldr) [inline bower.json]: indentation

File: package.json [hunk 5]
```diff
@@ -40,11 +40,10 @@
   ],
   "license": "MIT",
   "dependencies": {
```
Comment (heldr) [inline package.json]: I was wondering to change it to peerDependencies instead. What do you think @zenorocha?

Comment (heldr) [inline package.json]: I was wondering to change it to peerDependencies instead. What do you think @zenorocha?

**Comment by heldr** [comment]
@dcdeiv The idea of following **jQuery style guide** kinda make sense for this project. However more changes are also necessary:

1. Tests are failing because grunt-cli is missing. Please roolback it or install it through **travis.yml** ``before_script:`` instead.

2.  Make sure jslint is up to date with the new code style. Tests will fail otherwise.

File: bower.json [hunk 6]
```diff
@@ -1,18 +1,26 @@
 {
-	"name": "jquery-boilerplate",
-	"version": "4.0.0",
-	"homepage": "http://jqueryboilerplate.com",
+  "name": "jquery-boilerplate",
+  "version": "4.0.0",
```
Comment (heldr) [inline bower.json]: version is not needed here as per #109

**Comment by dcdeiv** [comment]
I reccommend not to remove **version** nor **license** in package.json nor in bower.json because when running bower install or npm install the shell will throw complain it hadn't find them!
So don't change them! Plus you should always have to provide a copy of the license in the repo!

File: LICENSE-MIT (hunk 1 above)
Comment (dcdeiv) [inline LICENSE-MIT]: Well, remove my name from the license and add yours! Licenses must always be provided!

File: bower.json (hunk 2 above)
Comment (dcdeiv) [inline bower.json]: removing the "license" from package.json and bower.json will throw error when the package will be installed via npm or Bower. Keep it with the copy of the license you must provide!

File: bower.json (hunk 4 above)
Comment (dcdeiv) [inline bower.json]: removing the "license" from package.json and bower.json will throw error when the package will be installed via npm or Bower. Keep it with the copy of the license you must provide!

**Comment by heldr** [comment]
Not always, a link to the license is enough!

BTW
https://github.com/jquery-boilerplate/jquery-boilerplate/pull/114/files#r53544356

**Comment by dcdeiv** [comment]
I added **JSCS**. As the [JSCS website](http://jscs.info/) says:

> JSCS is a code style linter/formatter for programmatically enforcing your style guide. You can configure JSCS for your project/company using over 150 validation rules, including presets from popular style guides like jQuery, Airbnb, Google, and more. 

This tool is used by the jQuery team to lint jQuery with the jQuery validation rules! So if you now run `grunt lint`, the task runner will pass the source to JShint and then to JSCS.

You can install [some plugins](http://jscs.info/overview#friendly-packages) too, so that your editor will says what you did wrong.

An example of output:

```shell
Running "jscs:src" (jscs) task
Missing space after opening round bracket at src/jquery.boilerplate.js :
     1 |// the semi-colon before function invocation is a safety net against concatenated
     2 |// scripts and/or other plugins which may not be closed properly.
     3 |;(function ( $, window, document, undefined ) {
----------^
     4 |
```
---

This build has failed all checks because of Karma. Karma has a lot of problems lately, yesterday I had problems installing it too and I ended up dropping it. But for this you should open a new issue!

**Comment by heldr** [comment]
Nice! :smile:  It solves some opened issues.

Could you please fix this version? https://github.com/jquery-boilerplate/jquery-boilerplate/pull/114/files#r53544356 and remove the license file (see the link on Readme.md bottom, there is no need for a file)

**Comment by dcdeiv** [comment]
:+1:

**Comment by dcdeiv** [comment]
I also removed the "moot" version from bower.json. Even when you `bower init` it does not add the version in the `bower.json` file.
#109

**Comment by heldr** [comment]
Thanks for the contribution, it will be released in version 4.1.0

**Pull request merged**